### PR TITLE
Allow SensorWatcher to create multiple mirrors of a remote sensor

### PR DESCRIPTION
**Pull request by bmerry**

### CODE REVIEW

File: src/aiokatcp/server.py [hunk 1]
```diff
@@ -475,7 +475,7 @@ def _service_task_done(self, task: asyncio.Task) -> None:
             except AttributeError:
                 # Python 3.7 does not have task names
                 name = "a task"
-            logger.warning("Halting the server because %s raised %s", name, exc)
+            logger.warning("Halting the server because %s raised %s", name, exc, exc_info=True)
```
Comment (bmerry) [inline src/aiokatcp/server.py:478]: Oops, this is an unrelated change I made to help with debugging service task failures.

File: tests/test_client.py [hunk 2]
```diff
@@ -744,29 +761,30 @@ def test_sensor_updated_unknown_sensor(self, watcher: DummySensorWatcher) -> Non
         self.test_sensor_added(watcher)
 
         watcher.batch_start()
-        watcher.sensor_updated("bar", b"123.0", Sensor.Status.WARN, 1234567890.0)
+        watcher.sensor_updated("spam", b"123.0", Sensor.Status.WARN, 1234567890.0)
         watcher.batch_stop()
         watcher.logger.warning.assert_called_once_with(  # type: ignore
-            "Received update for unknown sensor %s", "bar"
+            "Received update for unknown sensor %s", "spam"
```
Comment (james-smith-za) [inline tests/test_client.py:767]: For a Python company, we definitely don't have enough spam in our codebases.

**Review by james-smith-za — APPROVED** [review]
It ... looks fine, but I'm not sure what the use case is.

**Comment by bmerry** [comment]
> It ... looks fine, but I'm not sure what the use case is.

It's to allow katsdpcontroller to replicate fgpu per-engine sensors to each of the streams that the engine is handling.

**Pull request merged**